pkg/beholder/auth.go

```diff
@@ -17,11 +19,14 @@ var authHeaderVersion = "1"
 // <version>:<public_key_hex>:<signature_hex>
 //
 // where the byte value of <public_key_hex> is what's being signed
-func BuildAuthHeaders(privKey ed25519.PrivateKey) map[string]string {
-	pubKey := privKey.Public().(ed25519.PublicKey)
+func BuildAuthHeaders(ed25519Signer crypto.Signer) (map[string]string, error) {
```

Comment:
My original intent was to preserve existing calls that pass PKs, but there is an error to handle, so now the signature is broken. Should we introduce a new func instead? And just deprecate this one?